### DISCUSSION TITLE
Renaming initial flux-system git repository and kustomization removes controllers

### DISCUSSION THREAD

**Discussion by wvh** [fluxcd/flux2#3230] [General] [answered]
Hello!

I've tried renaming the initial `flux-system` git repository and kustomization after running `flux bootstrap` for a Github repository. This used to work at some point, but now it triggers removal of all controller pods from the `flux-system` namespace so Flux is uninstalled and nothing gets synchronised anymore. I imagine some objects hold a reference to other objects, and this rename causes a cascading delete instead of a wishful simple rename.

The reason I would like to rename those initial Git and Kustomization resources is because there are too many things called `flux-system` and it's confusing for people less familiar with Flux.

Is there a safe or recommended way to rename the bootstrapped Git and Kustomization objects? Alternatively, could there be a parameter for `flux bootstrap` to specify a different name?

Thanks!

**Comment by jirizaloudek** [comment]
The same question here. I would appreciate if I can use a real name of a Git repository instead of general flux-system.  

There is about 20 GitRepositories in my cluster and all of them are named same as the target Git repository except of `flux-system` what is a bit confusing.

**Comment by maulerjan** [comment]
Hi, any updates?

**Comment by mloskot** [comment]
I asked question that is partially related to this, see [Is it safe to rename Git repository specified in GitRepository?](https://github.com/fluxcd/flux2/discussions/5145)

**Comment by stefanprodan** [comment, marked as answer]
Renaming the root Flux Kustomization that `flux bootstrap` generates is not supported. 

If you really want a different name, you can run `flux uninstall --keep-namespace` and switch to [Flux Operator](https://github.com/controlplaneio-fluxcd/flux-operator) which allows setting a custom name of the root objects, docs here: https://fluxcd.control-plane.io/operator/fluxinstance/#sync-configuration
